mcux/hwmv2.cmake

```diff
@@ -0,0 +1,18 @@
+# When this module completes, the following variables will be defined:
+# MCUX_DEVICE: SOC name, suffixed by core name when using a dual core part.
+#              Example: MIMXRT595S_cm33, or LPC55S36
+# MCUX_CPU: "CPU"+ SOC part number, followed by core name when using a dual core part.
+#              Example: CPU_MIMXRT595SFAWC_cm33, or  CPU_LPC55S36JBD100
+# MCU_DEVICE_PATH: SOC name without core suffix. Must match the name of the
+#              folder in MCUX HAL. IE MIMXRT595S, or LPC55S36
+
+
+# For most SOCs, the below settings will suffice
+string(TOUPPER ${CONFIG_SOC} MCUX_DEVICE_PATH)
+string(TOUPPER ${CONFIG_SOC} MCUX_DEVICE)
+set(MCUX_CPU CPU_${CONFIG_SOC_PART_NUMBER})
+
+if(DEFINED CONFIG_MCUX_CORE_SUFFIX)
+  string(APPEND MCUX_DEVICE ${CONFIG_MCUX_CORE_SUFFIX})
```

Comment:
As "string(TOUPPER ${CONFIG_SOC} MCUX_DEVICE)", so CONFIG_SOC will not with CPU suffix, right? then CONFIG_SOC will have a single value and use CONFIG_MCUX_CORE_SUFFIX to differentiate CPU Core on this SoC.